### PR TITLE
Add registration of commitment tx's outputs from check_spend_remote_transaction

### PR DESCRIPTION
**Pull request by ariard**
Partial fulfilling of #137, simplest design of second suggestion ? Need to be tested when ChannelMonitor logic'll be completed.

### CODE REVIEW

File: src/ln/channelmonitor.rs [hunk 1]
```diff
@@ -464,8 +467,9 @@ impl ChannelMonitor {
 	/// optional, without it this monitor cannot be used in an SPV client, but you may wish to
 	/// avoid this (or call unset_funding_info) on a monitor you wish to send to a watchtower as it
 	/// provides slightly better privacy.
+	/// It is the responsability of the caller to register outpoint and script with passing the former 
+	/// value as key to update current monitor
 	pub(super) fn set_funding_info(&mut self, funding_info: (OutPoint, Script)) {
-		//TODO: Need to register the given script here with a chain_monitor
```
Comment (ariard) [inline src/ln/channelmonitor.rs]: Seems to me that registration of given script is already handled in internal_funding_created or funding_transaction_generated by call to add_update_monitor. If self.funding_txo is set, a call to add_update_monitor_by_key is enough.

**Comment by ariard** [comment]
"channel_monitor_network_test has been running for over 60 seconds" will investigate on tomorrow (and maybe add test)

File: src/ln/channelmonitor.rs [hunk 2]
```diff
@@ -69,7 +69,10 @@ impl<Key : Send + cmp::Eq + hash::Hash> ChainListener for SimpleManyChannelMonit
 	fn block_connected(&self, _header: &BlockHeader, height: u32, txn_matched: &[&Transaction], _indexes_of_txn_matched: &[u32]) {
 		let monitors = self.monitors.lock().unwrap();
 		for monitor in monitors.values() {
-			monitor.block_connected(txn_matched, height, &*self.broadcaster);
+			let outputs = monitor.block_connected(txn_matched, height, &*self.broadcaster);
+			for output in outputs {
+				self.chain_monitor.install_watch_script(&output.script_pubkey);
```
Comment (TheBlueMatt) [inline src/ln/channelmonitor.rs]: I think this needs to be install_watch_outpoint not script, as script was meant to monitor a specific tx by a random script, not meant to monitor outpoints which is all we need here (the commitment txn are already being monitored). Sorry for the confusion here, I had to go remind myself of what this stuff was for, heh.

Comment (ariard) [reply src/ln/channelmonitor.rs]: Ah yeah didn't get plainly the distinction! Thanks

**Comment by ariard** [comment]
I'd a review on your refactoring, I find it far more clearer this way with ChainWatchedUtil ! Just rebased your commits. 
Do you agree with my thoughts on first comment https://github.com/rust-bitcoin/rust-lightning/pull/159#discussion_r215826820 ?

**Comment by ariard** [comment]
Rebased, updated with I hope explicitly-enough comments!

**Pull request merged**